autoload/go/debug.vim

```diff
@@ -441,7 +457,8 @@ endfunction
 function! s:start_cb() abort
   let l:winid = win_getid()
   let l:debugwindows = go#config#DebugWindows()
-  if !empty(l:debugwindows)
+  let l:debugpreservelayout = go#config#DebugPreserveLayout()
+  if (empty(l:debugwindows) && l:debugpreservelayout == -1) || l:debugpreservelayout != 1
```

Comment:
`if empty(l:debugwindows) || l:debugpreservelayout`

Reply:
this is the condition for calling `only` ie not preserving the layout, so it needs to be flipped. i left off the `!` on `empty(` and fixed that tho